packages/amplify-go-function-runtime-provider/src/runtime.ts

```diff
@@ -0,0 +1,202 @@
+    throw new Error(`${executableName} failed, exit code was ${output.exitCode}`);
+  }
+
+  return output.stdout;
+};
+
+const isBuildStale = (resourceDir: string, lastBuildTimestamp: Date, outDir: string) => {
+  // If output directory does not exists or empty, rebuild required
+  if (!fs.existsSync(outDir) || glob.sync(`${outDir}/**`).length == 0) {
+    return true;
+  }
+
+  // If the timestamp of the src directory is newer than last build, rebuild required
+  const srcDir = path.join(resourceDir, SRC);
+  const dirTime = new Date(fs.statSync(srcDir).mtime);
+
+  if (dirTime > lastBuildTimestamp) {
+    return true;
+  }
+
+  const fileUpdatedAfterLastBuild = glob
+    .sync(`${resourceDir}/${SRC}/**`)
+    .find(file => new Date(fs.statSync(file).mtime) > lastBuildTimestamp);
+
+  return !!fileUpdatedAfterLastBuild;
+};
+
+export const buildResourceInternal = async (
+  request: BuildRequest,
+  context: any,
+  force: boolean,
+  forLocalInvoke: boolean,
+): Promise<BuildResult> => {
+  let rebuilt = false;
+
+  const buildDir = forLocalInvoke === true ? BIN_LOCAL : BIN;
+  const outDir = path.join(request.srcRoot, buildDir);
+
+  // For local invoke we've to use the build timestamp of the binary built
+  let timestampToCheck;
+
+  if (forLocalInvoke === true) {
+    const isWindows = /^win/.test(process.platform);
+    const executableName = isWindows === true ? MAIN_BINARY_WIN : MAIN_BINARY;
+    const executablePath = path.join(outDir, executableName);
+
+    if (fs.existsSync(executablePath)) {
+      timestampToCheck = new Date(fs.statSync(executablePath).mtime);
+    }
+  } else {
+    timestampToCheck = request.lastBuildTimestamp;
+  }
+
+  if (force === true || !timestampToCheck || isBuildStale(request.srcRoot, timestampToCheck, outDir)) {
+    const srcDir = path.join(request.srcRoot, SRC);
+    const entryFile = MAIN_SOURCE;
+    const isWindows = /^win/.test(process.platform);
+
+    // Clean and/or create the output directory
+    if (fs.existsSync(outDir)) {
+      fs.emptyDirSync(outDir);
+    } else {
+      fs.mkdirSync(outDir);
+    }
+
+    const envVars: any = {};
+
+    if (forLocalInvoke === false) {
+      envVars.GOOS = 'linux';
+      envVars.GOARCH = 'amd64';
+    }
+
+    if (isWindows) {
+      envVars.CGO_ENABLED = 0;
+    }
+
+    // Execute the build command, cwd must be the source file directory (Windows requires it)
+    executeCommand(['build', '-o', outDir, entryFile], true, envVars, srcDir);
+
+    rebuilt = true;
+  }
+
+  return {
+    rebuilt,
+  };
+};
+
+export const checkDependencies = async (_runtimeValue: string): Promise<CheckDependenciesResult> => {
+  // Check if go is in the path
+  executablePath = which.sync(executableName, {
+    nothrow: true,
+  });
+
+  if (executablePath === null) {
+    return {
+      hasRequiredDependencies: false,
+      errorMessage: `${executableName} executable was not found in PATH, make sure it's available or provide a custom path in the function.state file`,
```

Comment:
amplify.state file. However, we probably don't want to include that in the error message until we actually support that